src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -170,18 +179,30 @@ public Boolean replaceKibanaIndex(final ActionRequest request, final String acti
 
         }
 
+        return CONTINUE_EVALUATION_REPLACE_RESULT;
+    }
+
+    private CreateIndexRequest checkAndCreateRequestIfAbsent(final String name) {
+        final IndexAbstraction indexAbstraction = clusterService.state().getMetadata().getIndicesLookup().get(name);
+        if (indexAbstraction == null) {
+            return new CreateIndexRequest(name + "_1")
+                    .settings(KIBANA_INDEX_SETTINGS)
+                    .alias(new Alias(name));
+        }
+        log.debug("{} {} already exists", indexAbstraction.getType(), indexAbstraction.getName());
         return null;
     }
 
-    private void replaceIndex(final ActionRequest request, final String oldIndexName, final String newIndexName, final String action) {
+    private CreateIndexRequest replaceIndex(final ActionRequest request, final String oldIndexName, final String newIndexName, final String action) {
         boolean kibOk = false;
+        CreateIndexRequest createIndexRequest = null;
```

Comment:
All the request will be treated as CreateIndexRequest?

Reply:
`CreateIndexRequest` replaces auto create index request.